.github/workflows/compile-requirements.yml

```diff
@@ -0,0 +1,137 @@
+name: Reusable WF - Compile Pip Requirements
```

Comment:
Should the filename also say 'pip' in it? Or maybe this file should just be a general "compile requirements" even if now it is only pip focused? Could be more than that in the future.